R/aggregate.R

```diff
@@ -180,43 +143,380 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
     merged_dt[, (list_names) := lapply(list_names, function(x) {get(x) * w_area})]
   }
 
-  # Separate month and day columns
-  merged_dt[, ':=' (month = substring(date, first=1, last=8),
-                    day = substring(date, first=10))]
+  # Separate year, month, and day columns
+  merged_dt[, ':=' (year = substring(date, first = 1, last = 9),
+                    month = substring(date, first=10, last=17),
+                    day = substring(date, first=19))]
+
+  # Temporal Aggregation
+  if(agg_to == "year"){
+    sum_by_poly <- merged_dt[,  lapply(.SD, sum), by = .(poly_id, year),
+                             .SDcols = list_names]
 
-  # Can customize this in the future to aggregate by day & month
-  # Right now just sum by month
-  sum_by_poly <- merged_dt[,  lapply(.SD, sum), by = .(poly_id, month),
-                           .SDcols = list_names]
+    ## Order columns
+    data.table::setcolorder(sum_by_poly, neworder = c('year', 'poly_id', list_names))
+  }
+
+  if(agg_to == "month"){
+    sum_by_poly <- merged_dt[,  lapply(.SD, sum), by = .(poly_id, year, month),
+                             .SDcols = list_names]
+
+    ## Order columns
+    data.table::setcolorder(sum_by_poly, neworder = c('year', 'month', 'poly_id', list_names))
+  }
+
+  if(agg_to == "day"){
+    sum_by_poly <- merged_dt[,  lapply(.SD, sum), by = .(poly_id, year, month, day),
+                             .SDcols = list_names]
+
+    ## Order columns
+    data.table::setcolorder(sum_by_poly, neworder = c('year', 'month', 'day', 'poly_id', list_names))
+  }
 
-  ## Add year column
-  sum_by_poly[, year := year]
 
-  ## Order columns
-  data.table::setcolorder(sum_by_poly, neworder = c('year', 'month', 'poly_id', list_names))
 
   ## Return the sums by polygon
   return(sum_by_poly)
 
 
+
 }
 
-# function to call agg_climate_data over multiple years in parallel
-#' @describeIn agg_climate_data aggregate climate data over multiple years in parallel
-#' @param years A vector containing the years to aggregate over
 #' @export
-agg_climate_data_multiyear <- function(years, data_raster, climate_var, daily_agg, trans = 'polynomial', trans_specs, geoweights_table){
+staggregate_polynomial <- function(data_raster, climate_var, daily_agg, degree, geoweights_table, second_weights, agg_to = "month"){
+
+  # Get climate data as a data table and aggregate to daily values
+  setup_list <- before_trans(data_raster, climate_var, daily_agg, geoweights_table, second_weights)
+
+  clim_daily <- setup_list[[1]]
```

Comment:
for the record i do not love this solution as it requires hardcoding the order of objects in a list...and it's not clear in any comments what the order is. can we at least comment clearly what order the list returns different objects in?

Reply:
Yes I will make sure to describe objects and their order